src/neo-vm/Types/ByteString.cs

```diff
@@ -52,13 +55,15 @@ public override bool GetBoolean()
 
         public override int GetHashCode()
         {
-            unchecked
+            if (_hashCode == -1)
             {
-                int hash = 17;
-                foreach (byte element in GetSpan())
-                    hash = hash * 31 + element;
-                return hash;
+                unchecked
+                {
+                    using Murmur3 murmur = new((uint)GetSpan().Length);
```

Comment:
If we set `seed` to a random number after node start, that will be better.

The C# stdlib exactly this thing. <https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/HashCode.cs#L55>

Reply:
I don't like to introduce non determinism in the vm ... another `seed` idea?

Reply:
I think it's okay to use non-deterministic seed in `GetHashCode()`. Because it's not a part of the VM.